### PR TITLE
[#2727] refactor(core,server): refactor CatalogOperationDispatcher

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

Split CatalogOperationDispatcher into separate classes based on different operations.

### Why are the changes needed?

Fix: #2727 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

UTs

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java [hunk 1]
```diff
@@ -92,656 +61,7 @@ public CatalogOperationDispatcher(
...
-  @Override
-  public boolean dropFileset(NameIdentifier ident) {
-    return doWithCatalog(
-        getCatalogIdentifier(ident),
-        c -> c.doWithFilesetOps(f -> f.dropFileset(ident)),
-        NonEmptyEntityException.class);
-  }
-
-  private <R, E extends Throwable> R doWithCatalog(
+  <R, E extends Throwable> R doWithCatalog(
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java:64]: Why do you change to package private, not protected?

Comment (mchades) [reply core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java:64]: The `protected` modifier provides a wider scope of visibility than `package private`, do we need `protected`?

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java [hunk 2]
```diff
@@ -68,15 +37,15 @@
  * A catalog operation dispatcher that dispatches the catalog operations to the underlying catalog
  * implementation.
  */
-public class CatalogOperationDispatcher implements TableCatalog, FilesetCatalog, SupportsSchemas {
+public abstract class CatalogOperationDispatcher {
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java:40]: Can we rename this class to `OperationDispatcher`?

File: core/src/main/java/com/datastrato/gravitino/catalog/FilesetOperationDispatcher.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import com.datastrato.gravitino.file.Fileset;
+import com.datastrato.gravitino.file.FilesetCatalog;
+import com.datastrato.gravitino.file.FilesetChange;
+import com.datastrato.gravitino.storage.IdGenerator;
+import java.util.Map;
+
+public class FilesetOperationDispatcher extends CatalogOperationDispatcher
+    implements FilesetCatalog {
+  /**
+   * Creates a new CatalogOperationDispatcher instance.
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/catalog/FilesetOperationDispatcher.java:27]: The comment here seems not correct, should be `FilesetOperationDispatcher `

File: core/src/main/java/com/datastrato/gravitino/catalog/FilesetOperationDispatcher.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+   * @param store The EntityStore instance to be used for catalog operations.
+   * @param idGenerator The IdGenerator instance to be used for catalog operations.
+   */
+  public FilesetOperationDispatcher(
+      CatalogManager catalogManager, EntityStore store, IdGenerator idGenerator) {
+    super(catalogManager, store, idGenerator);
+  }
+
+  @Override
+  public NameIdentifier[] listFilesets(Namespace namespace) throws NoSuchSchemaException {
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/catalog/FilesetOperationDispatcher.java:39]: You'd better add javadoc for these methods.

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java (hunk 2 above)
Comment (mchades) [reply core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java]: done

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**